src/array.jl

```diff
@@ -396,14 +396,21 @@ copy(A::CategoricalArray) = deepcopy(A)
 CatArrOrSub{T, N} = Union{CategoricalArray{T, N},
                           SubArray{<:Any, N, <:CategoricalArray{T}}} where {T, N}
 
+function copy!(dest::CatArrOrSub, dstart::Integer,
```

Comment:
AFAICT we don't need to define this method, we can rely on the `AbstractArray` fallback. Just remove the default value of `n` in the method below.